src/manager.cpp

```diff
@@ -212,12 +212,91 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    if (i_vpdPath.empty())
+    {
+        logging::logMessage("Given VPD path is empty.");
+        return -1;
+    }
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    nlohmann::json l_sysCfgJsonObj{};
+
+    if (m_worker.get() != nullptr)
+    {
+        l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    int l_bytesUpdated = -1;
+    types::Path l_hwPath = i_vpdPath;
+    types::Path l_redundantFruPath;
+
+    try
+    {
+        if (!l_sysCfgJsonObj.empty())
+        {
+            // Get hardware path from system config JSON.
+            const std::string& l_fruPath =
+                jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_vpdPath);
+
+            if (!l_fruPath.empty())
+            {
+                // Save the FRU path to update on hardware
+                l_hwPath = l_fruPath;
+
+                // Get inventory object path from system config JSON
+                const std::string& l_inventoryObjPath =
+                    jsonUtility::getInventoryObjPathFromJson(l_sysCfgJsonObj,
+                                                             i_vpdPath);
+
+                if (!l_inventoryObjPath.empty())
+                {
+                    // TODO: Extract the record, keyword, and value from input
+                    // tuple
+                    // TODO: Read existing value from EEPROM by passing record
+                    // and keyword
+                    // TODO: Compare the given keyword's size vs the size of
+                    // keyword read from EEPROM
+                    // TODO: Update keyword's value on D-bus with right size
+                    // TODO: Log a message if update is successful on D-bus
+                }
+
+                // Get redundant hardware path if present in system config JSON
+                l_redundantFruPath =
+                    jsonUtility::getRedundantEepromPathFromJson(l_sysCfgJsonObj,
+                                                                i_vpdPath);
+            }
+        }
+
+        // Update keyword's value on hardware
+        l_bytesUpdated = updateKeywordOnHardware(l_hwPath, l_sysCfgJsonObj,
+                                                 i_paramsToWriteData);
+
+        if (l_bytesUpdated == -1)
+        {
+            // TODO: Revert D-bus update
+
+            return l_bytesUpdated;
+        }
+
+        logging::logMessage("Keyword update successful on " + l_hwPath);
+
+        // Update keyword's value on redundant hardware if present
+        if (!l_redundantFruPath.empty())
```

Comment:
why this is here and not within the frupath check ?  move this to line 265 please

Reply:
Only if update on primary EEPROM succeeds, we should try update on redundant eeprom (as redundant acts as backup for primary). if update on primary fails, stop there and don't try to update the backup.

Reply:
as discussed with Jinu, moved the redundant update before hardware update. as the code piece which deals with JSON has to be clubbed together.